### ISSUE TITLE
viu.com cannot be downloaded with other downloaders

### ISSUE THREAD

**Issue by ltsj** [yt-dlp/yt-dlp#4925]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.09.01** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Hong Kong(Including Singapore and other countries where it provides services)

### Provide a description that is worded well enough to be understood

`yt-dlp -vU -o "%(episode_number)s"  https://www.viu.com/ott/hk/zh-hk/vod/443782/%E5%85%AD%E6%9C%AC%E6%9C%A8Class -I 1 --sub-langs all  --sub-format "srt" --write-sub  --downloader aria2c --downloader-args aria2c:"-x 16 -k 10M"`
The above is the command I use. I want to download it through aria2, but it will not work. The same is true of other downloaders. I have tried other websites, and other downloaders can be used normally.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', '-o', '%(episode_number)s', 'https://www.viu.com/ott/hk/zh-hk/vod/443782/%E5%85%AD%E6%9C%AC%E6%9C%A8Class', '-I', '1', '--sub-langs', 'all', '--sub-format', 'srt', '--write-sub', '--downloader', 'aria2c', '--downloader-args', 'aria2c:-x 16 -k 10M']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.09.01 [5d7c7d6] (linux_exe)
[debug] Python 3.10.5 (CPython 64bit) - Linux-5.10.0-11-amd64-x86_64-with-glibc2.31 (glibc 2.31)
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg N-61656-gfbd22504c4-static (setts), phantomjs 139760848758720
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.06.15, mutagen-1.45.1, secretstorage-3.3.3, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Loaded 1670 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.09.01, Current version: 2022.09.01
yt-dlp is up to date (2022.09.01)
[debug] [viu:ott] Extracting URL: https://www.viu.com/ott/hk/zh-hk/vod/443782/%E5%85%AD%E6%9C%AC%E6%9C%A8Class
[viu:ott] 443782: Downloading video info
[viu:ott] Downloading playlist 24034 - add --no-playlist to download just the video 443782
[download] Downloading playlist: 六本木Class
[viu:ott] Playlist 六本木Class: Downloading 1 videos of 10
[download] Downloading video 1 of 1
[debug] [viu:ott] Extracting URL: http://www.viu.com/ott/hk/zh-hk/vod/443782/#__youtubedl_smuggle=%7B%22force_noplaylist%22%3A+true%7D
[viu:ott] 443782: Downloading video info
[viu:ott] 443782: Getting bearer token
[viu:ott] 443782: Downloading stream info
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] 443782: Downloading subtitles: 繁體中文
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 443782: Downloading 1 format(s): s1080p
[info] Writing video subtitles to: 1.繁體中文.srt
[debug] Invoking aria2c downloader on "https://d2anahhhmp1ffz.cloudfront.net/2454472826/0b7f1d7336f4880d6b4bcf8765a2d3b2aa1a8deb"
[download] Destination: 1.繁體中文.srt
[debug] aria2c command line: aria2c -c --console-log-level=warn --summary-interval=0 --download-result=hide --http-accept-gzip=true --file-allocation=none -x16 -j16 -s16 --min-split-size 1M --header 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.41 Safari/537.36' --header 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8' --header 'Accept-Language: en-us,en;q=0.5' --header 'Sec-Fetch-Mode: navigate' --header 'Cookie: dtCookie=v_4_srv_7_sn_4021BD2D8E6554449073552E756BD249_perc_100000_ol_0_mul_1_app-3A4e1152bec816f955_0' --check-certificate=true --remote-time=true --show-console-readout=true -x 16 -k 10M --out './1.繁體中文.srt.part' --auto-file-renaming=false -- https://d2anahhhmp1ffz.cloudfront.net/2454472826/0b7f1d7336f4880d6b4bcf8765a2d3b2aa1a8deb
[aria2c] Downloaded 41608 bytes
[download] 100% of 40.63KiB in 00:00 at 604.80KiB/s
[debug] Invoking ffmpeg downloader on "https://stream-hk.viu.com/s/mM8nSuS2xdSClqosVjJnTA/1663192528/UD/286ece264697ce824569abeba698f326/286ece264697ce824569abeba698f326_Layer4_vod.m3u8"
[download] Destination: 1
[debug] ffmpeg command line: ffmpeg -y -loglevel verbose -headers 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.41 Safari/537.36
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Language: en-us,en;q=0.5
Sec-Fetch-Mode: navigate
Cookie: dtCookie=v_4_srv_7_sn_4021BD2D8E6554449073552E756BD249_perc_100000_ol_0_mul_1_app-3A4e1152bec816f955_0
' -i https://stream-hk.viu.com/s/mM8nSuS2xdSClqosVjJnTA/1663192528/UD/286ece264697ce824569abeba698f326/286ece264697ce824569abeba698f326_Layer4_vod.m3u8 -c copy -f mp4 file:1.part
ffmpeg version N-61656-gfbd22504c4-static https://johnvansickle.com/ffmpeg/  Copyright (c) 2000-2022 the FFmpeg developers
  built with gcc 8 (Debian 8.3.0-6)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-debug --disable-ffplay --disable-indev=sndio --disable-outdev=sndio --cc=gcc --enable-fontconfig --enable-frei0r --enable-gnutls --enable-gmp --enable-libgme --enable-gray --enable-libaom --enable-libfribidi --enable-libass --enable-libvmaf --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librubberband --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libvorbis --enable-libopus --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libdav1d --enable-libxvid --enable-libzvbi --enable-libzimg
  libavutil      57. 24.101 / 57. 24.101
  libavcodec     59. 26.100 / 59. 26.100
  libavformat    59. 22.100 / 59. 22.100
  libavdevice    59.  6.100 / 59.  6.100
  libavfilter     8. 33.100 /  8. 33.100
  libswscale      6.  6.100 /  6.  6.100
  libswresample   4.  6.100 /  4.  6.100
  libpostproc    56.  5.100 / 56.  5.100
[tcp @ 0x5ce0ec0] Starting connection attempt to 203.135.138.10 port 443
[tcp @ 0x5ce0ec0] Successfully connected to 203.135.138.10 port 443
[hls @ 0x5cdcb40] Skip ('#EXT-X-VERSION:3')
[hls @ 0x5cdcb40] Skip ('#EXT-X-ALLOW-CACHE:YES')
[hls @ 0x5cdcb40] HLS request for url 'https://stream-hk.viu.com/s/mM8nSuS2xdSClqosVjJnTA/1663192528/UD/286ece264697ce824569abeba698f326/286ece264697ce824569abeba698f326_Layer4/Period1/live_00000000.ts', offset 0, playlist 0
[hls @ 0x5cdcb40] Opening 'https://stream-hk.viu.com/hls/vodcp124/vodcp124.key' for reading
[tcp @ 0x5ff3800] Starting connection attempt to 203.135.138.10 port 443
[tcp @ 0x5ff3800] Successfully connected to 203.135.138.10 port 443
[AVIOContext @ 0x62ee280] Statistics: 16 bytes read, 0 seeks
[hls @ 0x5cdcb40] Opening 'crypto+https://stream-hk.viu.com/s/mM8nSuS2xdSClqosVjJnTA/1663192528/UD/286ece264697ce824569abeba698f326/286ece264697ce824569abeba698f326_Layer4/Period1/live_00000000.ts' for reading
[tcp @ 0x5fbfe80] Starting connection attempt to 203.135.138.10 port 443
[tcp @ 0x5fbfe80] Successfully connected to 203.135.138.10 port 443
[NULL @ 0x60eabc0] non-existing SPS 0 referenced in buffering period
[h264 @ 0x60017c0] non-existing SPS 0 referenced in buffering period
[h264 @ 0x60017c0] Reinit context to 1920x1088, pix_fmt: yuv420p
Input #0, hls, from 'https://stream-hk.viu.com/s/mM8nSuS2xdSClqosVjJnTA/1663192528/UD/286ece264697ce824569abeba698f326/286ece264697ce824569abeba698f326_Layer4_vod.m3u8':
  Duration: 00:53:51.88, start: 1.437344, bitrate: 0 kb/s
  Program 0 
    Metadata:
      variant_bitrate : 0
  Stream #0:0: Video: h264 (High), 1 reference frame ([27][0][0][0] / 0x001B), yuv420p(left), 1920x1080 (1920x1088) [SAR 1:1 DAR 16:9], 25 fps, 25 tbr, 90k tbn
    Metadata:
      variant_bitrate : 0
  Stream #0:1: Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 0
Output #0, mp4, to 'file:1.part':
  Metadata:
    encoder         : Lavf59.22.100
  Stream #0:0: Video: h264 (High), 1 reference frame (avc1 / 0x31637661), yuv420p(left), 1920x1080 (0x0) [SAR 1:1 DAR 16:9], q=2-31, 25 fps, 25 tbr, 90k tbn
    Metadata:
      variant_bitrate : 0
  Stream #0:1: Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 0
Stream mapping:
  Stream #0:0 -> #0:0 (copy)
  Stream #0:1 -> #0:1 (copy)
Press [q] to stop, [?] for help
Automatically inserted bitstream filter 'aac_adtstoasc'; args=''ate=N/A speed=   0x    
[AVIOContext @ 0x6007140] Statistics: 6504048 bytes read, 0 seekste=4426.1kbits/s speed=4.87x    
[hls @ 0x5cdcb40] HLS request for url 'https://stream-hk.viu.com/s/mM8nSuS2xdSClqosVjJnTA/1663192528/UD/286ece264697ce824569abeba698f326/286ece264697ce824569abeba698f326_Layer4/Period1/live_00000001.ts', offset 0, playlist 0
[hls @ 0x5cdcb40] Opening 'crypto+https://stream-hk.viu.com/s/mM8nSuS2xdSClqosVjJnTA/1663192528/UD/286ece264697ce824569abeba698f326/286ece264697ce824569abeba698f326_Layer4/Period1/live_00000001.ts' for reading
[tcp @ 0x6468800] Starting connection attempt to 203.135.138.10 port 443
[tcp @ 0x6468800] Successfully connected to 203.135.138.10 port 443
[mpegts @ 0x5fe70c0] Packet corrupt (stream = 0, dts = 1054800).
[hls @ 0x5cdcb40] Packet corrupt (stream = 0, dts = 1051200).
https://stream-hk.viu.com/s/mM8nSuS2xdSClqosVjJnTA/1663192528/UD/286ece264697ce824569abeba698f326/286ece264697ce824569abeba698f326_Layer4_vod.m3u8: corrupt input packet in stream 0
[AVIOContext @ 0x62420c0] Statistics: 6237276 bytes read, 0 seekste=4475.0kbits/s speed=2.44x    
[hls @ 0x5cdcb40] HLS request for url 'https://stream-hk.viu.com/s/mM8nSuS2xdSClqosVjJnTA/1663192528/UD/286ece264697ce824569abeba698f326/286ece264697ce824569abeba698f326_Layer4/Period1/live_00000002.ts', offset 0, playlist 0
[hls @ 0x5cdcb40] Opening 'crypto+https://stream-hk.viu.com/s/mM8nSuS2xdSClqosVjJnTA/1663192528/UD/286ece264697ce824569abeba698f326/286ece264697ce824569abeba698f326_Layer4/Period1/live_00000002.ts' for reading
[tcp @ 0x62449c0] Starting connection attempt to 203.135.138.10 port 443
[tcp @ 0x62449c0] Successfully connected to 203.135.138.10 port 443
[mpegts @ 0x5fe70c0] Packet corrupt (stream = 0, dts = 1983600).
[hls @ 0x5cdcb40] Packet corrupt (stream = 0, dts = 1980000).
https://stream-hk.viu.com/s/mM8nSuS2xdSClqosVjJnTA/1663192528/UD/286ece264697ce824569abeba698f326/286ece264697ce824569abeba698f326_Layer4_vod.m3u8: corrupt input packet in stream 0
```

**Comment by pukkandan** [comment]
Can't reproduce. Was it a live stream when you were trying it? Live streams can only be downloaded with ffmpeg

**Comment by ltsj** [comment]
It's not a live broadcast, it's a series . I tried to use the extracted m3u8 link(https://stream-hk.viu.com/s/mM8nSuS2xdSClqosVjJnTA/1663192528/UD/286ece264697ce824569abeba698f326/286ece264697ce824569abeba698f326_Layer4_vod.m3u8) to download, it can be downloaded using aria2.

**Comment by pukkandan** [comment]
my bad, I couldn't repro since I was testing with master and this has already been fixed by deae7c171180ddd4735c414306f084f86ef27e07
